ipywidgets/widgets/embed.py

```diff
@@ -0,0 +1,262 @@
+        The state to include. When set to None, the state of all widgets
+        know to the widget manager is included. Otherwise it uses the
+        passed state directly. This allows for end users to include a
+        smaller state, under the responsibility that this state is
+        sufficient to reconstruct the embedded views.
+
+    Returns
+    -------
+    A dictionary with the following entries:
+        manager_state: dict of the widget manager state data
+        view_specs: a list of widget view specs
+    """
+    if views is None:
+        views = [w for w in Widget.widgets.values() if isinstance(w, DOMWidget)]
+    else:
+        try:
+            views[0]
+        except (IndexError, TypeError):
+            views = [views]
+
+    if state is None:
+        # Get state of all known widgets
+        state = state = Widget.get_manager_state(drop_defaults=drop_defaults, widgets=None)['state']
+
+    # Rely on ipywidget to get the default values
+    json_data = Widget.get_manager_state(widgets=[])
+    # but plug in our own state
+    json_data['state'] = state
+
+    view_specs = [w.get_view_spec() for w in views]
+
+    return dict(manager_state=json_data, view_specs=view_specs)
+
+
+def embed_snippet(views,
+                  drop_defaults=True,
+                  state=None,
+                  indent=2,
+                  embed_url=None,
+                 ):
+    """Return a snippet that can be embedded in an HTML file.
+
+    Parameters
+    ----------
+    views: widget or collection of widgets or None
+        The widgets to include views for. If None, all DOMWidgets are
+        included (not just the displayed ones).
+    drop_defaults: boolean
+        Whether to drop default values from the widget states.
+    state: dict or None (default)
+        The state to include. When set to None, the state of all widgets
+        know to the widget manager is included. Otherwise it uses the
+        passed state directly. This allows for end users to include a
+        smaller state, under the responsibility that this state is
+        sufficient to reconstruct the embedded views.
+    indent: integer, string or None
+        The indent to use for the JSON state dump. See `json.dumps` for
+        full description.
+    embed_url: string or None
+        Allows for overriding the URL used to fetch the widget manager
+        for the embedded code. This defaults (None) to an `unpkg` CDN url.
+
+    Returns
+    -------
+    A unicode string with an HTML snippet containing several `<script>` tags.
+    """
+
+    data = embed_data(views, drop_defaults=drop_defaults, state=state)
+
+    widget_views = u'\n'.join(
+        widget_view_template.format(**dict(view_spec=json.dumps(view_spec)))
+        for view_spec in data['view_specs']
+    )
+
+    if embed_url is None:
+        # TODO: Get widgets npm version automatically:
+        embed_url = u'https://unpkg.com/jupyter-js-widgets@~3.0.0-alpha.0/dist/embed.js'
```

Comment:
PS: We could probably have a function (in `_version.py`?) that peeks in the `package.json` files of the packages.